tests/operators/test_subdag_operator.py

```diff
@@ -237,3 +240,65 @@ def test_rerun_failed_subdag(self):
 
         sub_dagrun.refresh_from_db()
         self.assertEqual(sub_dagrun.state, State.RUNNING)
+
+    @parameterized.expand([
+        (SkippedStatePropagationOptions.ALL_LEAVES, [State.SKIPPED, State.SKIPPED], True),
+        (SkippedStatePropagationOptions.ALL_LEAVES, [State.SKIPPED, State.SUCCESS], False),
+        (SkippedStatePropagationOptions.ANY_LEAF, [State.SKIPPED, State.SUCCESS], True),
+        (SkippedStatePropagationOptions.ANY_LEAF, [State.FAILED, State.SKIPPED], True),
```

Comment:
I think this might already be covered by other tests but maybe we should add `None, [State.FAILED, State. FAILED]` ! and check if `FAILED` state

Reply:
I will add this one:
```python
(None, [State.SKIPPED, State.SKIPPED], False),
```

I am only testing for dag_run's which succeeded.

So 
```python
(None, [State.FAILED, State. FAILED], False),
```
should not be possible in a succeeded dag run, right?

Failed dag_run's will raise an exception before even coming to the new functionality I added.
```python
if dag_run.state != State.SUCCESS:
    raise AirflowException(
        "Expected state: SUCCESS. Actual state: {}".format(dag_run.state)
    )
```

Should I also test for failed dag_run's ?

Reply:
Failed dag_run's are covered here https://github.com/apache/airflow/pull/6352/files#diff-110fdb12f5df15ebf205dca824886e3aR172-R189

Reply:
Yup you are right, just add `(None, [State.SKIPPED, State.SKIPPED], False),
`